src/main/java/fr/xephi/authme/process/email/AsyncAddEmail.java

```diff
@@ -57,14 +62,19 @@ public void addEmail(Player player, String email) {
             } else if (!validationService.isEmailFreeForRegistration(email, player)) {
                 service.send(player, MessageKey.EMAIL_ALREADY_USED_ERROR);
             } else {
+                EmailChangedEvent event = bukkitService.createAndCallEvent(isAsync
+                    -> new EmailChangedEvent(player, null, email, isAsync));
+                if (event.isCancelled()) {
+                    sendFailedMessage(player);
```

Comment:
In my opinion, this makes it hard to track down errors. `dataSource.updateEmail` will print an exception if an error occurrs, but if the event is cancelled you only get a very abstract error message.

My suggestion is to use a different error message for the console where you can see it's because the event was cancelled.